contracts/protocol/pool/PoolApeStaking.sol

```diff
@@ -305,11 +314,38 @@ contract PoolApeStaking is
 
         // 2, send cash part to xTokenAddress
         if (stakingInfo.cashAmount > 0) {
-            APE_COIN.safeTransferFrom(
-                msg.sender,
-                localVar.xTokenAddress,
-                stakingInfo.cashAmount
-            );
+            if (stakingInfo.cashAsset == address(APE_COIN)) {
+                APE_COIN.safeTransferFrom(
+                    msg.sender,
+                    localVar.xTokenAddress,
+                    stakingInfo.cashAmount
+                );
+            } else {
+                DataTypes.ReserveData storage cApeReserve = ps._reserves[
+                    address(APE_COMPOUND)
+                ];
+                DataTypes.ReserveCache memory cApeReserveCache = cApeReserve
+                    .cache();
+                cApeReserve.updateState(cApeReserveCache);
+                cApeReserve.updateInterestRates(
+                    cApeReserveCache,
+                    address(APE_COMPOUND),
+                    0,
+                    stakingInfo.cashAmount
+                );
+                IPToken(PCAPE).burn(
```

Comment:
since we already read cApeReserveCache, why not we just use xTokenAddress from that?

Reply:
yes，we can.